Source/Core/ArcGisImageServerTerrainProvider.js

```diff
@@ -152,6 +152,18 @@ define([
         },
 
         /**
+         * Gets a promise that resolves to true when the provider is ready for use.
+         * @memberof ArcGisImageServerTerrainProvider.prototype
+         * @type {Promise.<Boolean>}
+         * @readonly
+         */
+        readyPromise : {
+            get : function() {
+                return when.resolve(true);
```

Comment:
You should define `this._readyPromise = when.resolve(true);` in the constructor and then simply return it here.  The code as it is now returns a different promise every time.